plugins/panels/model_evaluation.py

```diff
@@ -585,6 +610,89 @@ def load_view(self, ctx):
                     view = eval_view.filter_labels(
                         pred_field, F(eval_key) == field, only_matches=True
                     )
+        elif info.config.type == "segmentation":
+            results = ctx.dataset.load_evaluation_results(eval_key)
+            classes_map = self.get_classes_map(ctx.dataset, results, gt_field)
+            if (
+                results.ytrue_ids_dict is None
+                or results.ypred_ids_dict is None
+            ):
+                # legacy segmentation evaluation
+                return
+
+            if eval_key2:
+                if gt_field2 is None:
+                    gt_field2 = gt_field
+
+                results2 = ctx.dataset.load_evaluation_results(eval_key2)
+                classes_map2 = self.get_classes_map(
+                    ctx.dataset, results2, gt_field2
+                )
+                if (
+                    results2.ytrue_ids_dict is None
+                    or results2.ypred_ids_dict is None
+                ):
+                    # legacy segmentation evaluation
+                    return
+            else:
+                results2 = None
+
+            _, gt_id = ctx.dataset._get_label_field_path(gt_field, "_id")
+            _, pred_id = ctx.dataset._get_label_field_path(pred_field, "_id")
+            if gt_field2 is not None:
+                _, gt_id2 = ctx.dataset._get_label_field_path(gt_field2, "_id")
+            if pred_field2 is not None:
+                _, pred_id2 = ctx.dataset._get_label_field_path(
+                    pred_field2, "_id"
+                )
+
+            if view_type == "class":
+                # All GT/predictions that contain class `x`
+                k = classes_map[x]
+                ytrue_ids, ypred_ids = _get_ids_slice(results, k)
```

Comment:
These are the callbacks that fire when a user clicks on a cell in the `Class performance` section. The current data model requires building up a list of IDs for entire row/column slices of the ID dicts. We could potentially optimize the data model for this case. Needs some more thought.